### DISCUSSION TITLE
Vertipaq Analyzer via Advanced Scripting

### DISCUSSION THREAD

**Discussion by m-kovalsky** [TabularEditor/TabularEditor3#1060] [Ideas]
Following up from this issue (https://github.com/TabularEditor/PublicPreview/issues/90)

* (When) will we get access to reference additional vertipaq stats via Advanced Scripting (besides cardinality)?
* The 'GetCardinality()' function outputs into an 'information pop-up'. Is it possible to use this function to store the output value in a variable? Or does this require a different function?

**Comment by otykier** [comment]
Today, you can use the following methods:

```csharp
var tableCardinality = Model.Tables["Internet Sales"].GetCardinality(); // Gets the table row count
var tableSize = Model.Tables["Internet Sales"].GetTotalSize(); // Gets the size of the table in bytes
var columnCardinality = Model.Tables["Internet Sales"].Columns["Sales Amount"].GetCardinality(); // Gets the number of unique values in the column
var columnSize = Model.Tables["Internet Sales"].Columns["Sales Amount"].GetTotalSize(); // Gets the total size of the column in bytes
````

What other stats are you looking for?

I don't follow your 2nd comment - the GetCardinality() and GetTotalSize() methods both return a `long` value. No information popup.

**Comment by m-kovalsky** [comment]
Ah got it, thanks. Sorry about the 2nd comment - my bad!

Other stats which would be helpful:

Partitions: RecordCount, SegmentCount, RecordsPerSegment
Columns: HierarchySize, DataSize, DictionarySize
Tables: DataSize, DictionarySize, RI Violations
Hierarchies: UserHierarchySize
Relationship: RelationshipSize, MaxFromCardinality, MaxToCardinality, MissingKeys, InvalidRows
